src/nvim/regexp.c

```diff
@@ -4941,16 +4974,16 @@ static int regmatch(
                      < behind_pos.rs_u.pos.lnum
                      ? (colnr_T)STRLEN(regline)
                      : behind_pos.rs_u.pos.col)
-                    - rp->rs_un.regsave.rs_u.pos.col >= limit))
+                    - rp->rs_un.regsave.rs_u.pos.col >= limit)) {
               no = FAIL;
-            else if (rp->rs_un.regsave.rs_u.pos.col == 0) {
+            } else if (rp->rs_un.regsave.rs_u.pos.col == 0) {
               if (rp->rs_un.regsave.rs_u.pos.lnum
                   < behind_pos.rs_u.pos.lnum
                   || reg_getline(
-                      --rp->rs_un.regsave.rs_u.pos.lnum)
-                  == NULL)
+                      (rp--)->rs_un.regsave.rs_u.pos.lnum)
```

Comment:
increment changed here. not sure if semantics are preserved

Reply:
this is wrong. Did the linter actually complain about this line?

Reply:
I can omit all increment related changes, leaving only braces and spaces

Reply:
I reverted this one